tests/30rooms/05aliases.pl

```diff
@@ -226,6 +226,37 @@ sub _test_can_create_and_delete_alias {
       })
    };
 
+test "Users with sufficient power-level can delete other's aliases",
+   requires => [ $creator_fixture, $room_fixture, local_user_fixture(), room_alias_fixture(),
+                 qw( can_create_room_alias )],
+
+   do => sub {
+      my ( $user, $room_id, $other_user, $room_alias ) = @_;
+      my $server_name = $user->http->server_name;
+
+      matrix_change_room_power_levels(
+         $user, $room_id, sub {
+            $_[0]->{users}->{$other_user->user_id} = 100;
+         },
+      )->then( sub {
+         do_request_json_for( $user,
+           method => "PUT",
+           uri    => "/r0/directory/room/$room_alias",
+
+           content => { room_id => $room_id },
+         )
+      })->then( sub {
+         do_request_json_for( $other_user,
```

Comment:
shouldn't `other_user` be a member of the room?

Reply:
It doesn't seem to be a requirement. We can have the user join though if you think it is best.

Reply:
This seems to overlap with the comments at https://github.com/matrix-org/synapse/pull/6986#discussion_r384598785. We should sync about the expected behavior.

Reply:
I added this check!